### PR TITLE
feat: add rate limiting middleware (Issue #92)

### LINKED ISSUE #92: Add rate limiting middleware for API protection
## Summary

Implement rate limiting to protect the API from brute force attacks, denial of service, and credential stuffing when deployed internet-facing on Cloud Foundry.

## Context

Deferred from issue #44 (JWT signature verification) design session. See `docs/plans/2026-01-30-jwt-signature-verification-design.md` for context.

### Endpoints Requiring Rate Limiting

| Endpoint Category | Suggested Limit | Rationale |
|-------------------|-----------------|-----------|
| `/auth/login` | 5 req/min per IP | Brute force prevention |
| `/auth/refresh` | 10 req/min per session | Token abuse prevention |
| Planning endpoints | 10 req/min per user | Expensive calculations |
| Read endpoints | 100 req/min per user | General protection |

### Architecture Decision Needed

Evaluate these approaches:

1. **In-process** - Go middleware with in-memory counters. Simple, but resets on restart and doesn't work across instances.

2. **Redis-backed** - Distributed rate limiting. Works across instances. Requires Redis service binding.

3. **Platform-level** - CF route services, Istio, or API gateway. Keeps app simple but adds infrastructure dependency.

## Acceptance Criteria

- [ ] Document architecture decision with trade-offs
- [ ] Implement chosen rate limiting approach
- [ ] Apply stricter limits to authentication endpoints
- [ ] Return standard 429 Too Many Requests with Retry-After header
- [ ] Add tests for rate limiting behavior
- [ ] Document rate limits in OpenAPI spec

## Related

- #44: JWT signature verification (parent design session)
- Design doc: `docs/plans/2026-01-30-jwt-signature-verification-design.md`

### PR DESCRIPTION
**Pull request by malston**
## Summary

- Add per-endpoint rate limiting middleware with fixed-window counters to protect against brute force, credential stuffing, and API abuse
- Implement four rate limit tiers: auth (5/min), refresh (10/min), write (10/min), default (100/min), plus exempt tier for health/openapi
- Key extraction by client IP (X-Forwarded-For), session cookie, or user identity with automatic fallback
- All limits configurable via environment variables, entire system disableable via `RATE_LIMIT_ENABLED=false`

Closes #92

## Review feedback addressed

- Fix memory leak: lazy deletion of expired entries + periodic sweep every 100 new windows
- Document XFF trust assumption on ClientIP (safe behind CF gorouter)
- Change rate limit log level from Debug to Warn for security monitoring
- Add config validation: rate limit values must be 1-10000
- Add tests: empty XFF header, concurrent multi-key access, expired entry cleanup, invalid config values

## Test plan

- [x] 18 unit tests: core limiter (within/over limit, separate keys, window reset, concurrency, multi-key concurrency, expired cleanup), key extractors (ClientIP, SessionKey, UserOrIP, empty XFF), middleware factory (nil limiter, empty key, 429 response, Content-Type)
- [x] 5 E2E tests: login rate limiting (5 OK, 6th returns 429), exempt endpoints, disabled mode, separate IP quotas, write endpoint limiting
- [x] 5 config tests: default values, environment variable overrides, invalid value validation (zero, negative, exceeds max)
- [x] All 250 frontend tests pass
- [x] Linter clean (staticcheck + eslint)
- [x] Manual: start server, hit `/api/v1/auth/login` 6 times rapidly, verified 429 on 6th with `Retry-After: 60` header and JSON body

### CODE REVIEW

File: backend/middleware/ratelimit.go [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+}
+
+// RateLimiter enforces a maximum number of requests per time window.
+// Each unique key (IP, user, session) gets an independent counter.
+type RateLimiter struct {
+	mu      sync.Mutex
+	windows map[string]*counter
+	limit   int
+	window  time.Duration
+}
```
[the remark this answers is not in the document]
Comment (malston) [reply backend/middleware/ratelimit.go]: Fixed in dda8a1e (prior review round). Added lazy deletion of expired entries when a key is re-accessed, plus a periodic sweep every 100 new windows to bound stale entries. See `Allow()` lines 54-68.

File: backend/config/config.go [hunk 2]
```diff
@@ -61,6 +68,12 @@ func Load() (*Config, error) {
 		CORSAllowedOrigins: getEnvStringList("CORS_ALLOWED_ORIGINS"),
 		CookieSecure:       getEnvBool("COOKIE_SECURE", true),
 
+		RateLimitEnabled: getEnvBool("RATE_LIMIT_ENABLED", true),
+		RateLimitAuth:    getEnvInt("RATE_LIMIT_AUTH", 5),
+		RateLimitRefresh: getEnvInt("RATE_LIMIT_REFRESH", 10),
+		RateLimitWrite:   getEnvInt("RATE_LIMIT_WRITE", 10),
+		RateLimitDefault: getEnvInt("RATE_LIMIT_DEFAULT", 100),
+
```
[the remark this answers is not in the document]
Comment (malston) [reply backend/config/config.go]: Fixed in dda8a1e (prior review round). Added validation loop that rejects values outside 1-10000 range with a descriptive error. Tests added for zero, negative, and exceeds-max cases in `TestLoadConfig_RateLimitInvalidValue`.
